src/Compatibility/Core/src/RendererToHandlerShim.cs

```diff
@@ -85,14 +98,15 @@ protected override void DisconnectHandler(global::Android.Views.View nativeView)
 
 		public override void SetVirtualView(IView view)
 		{
-			if (VisualElementRenderer == null && Context != null)
+			if (VisualElementRenderer == null)
 			{
-				var renderer = Internals.Registrar.Registered.GetHandlerForObject<IVisualElementRenderer>(view, Context)
-										   ?? new Platform.Android.AppCompat.Platform.DefaultRenderer(Context);
-
-				SetupRenderer(renderer);
+				SetupRenderer(CreateRenderer(view));
 			}
 
+			SetRenderer(
```

Comment:
This is the main code change from the PR. Just moving this setter up before `SetElement` is called